### PR TITLE
Feature/app 2826 inss fip pipelines

### PR DESCRIPTION
**Pull request by kwan-nomensa**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
+trigger:
+- release/*
```
Comment (eman168) [inline azure-pipelines.yml:2]: should this include "develop"?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      - task: DotNetCoreCLI@2
+        displayName: 'Publish Project'
+        inputs:
+          command: 'publish'
+          publishWebProjects: false
+          projects: '**/*.csproj'
+          arguments: '--configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)'
+        condition: eq('${{ parameters.doPublish}}', 'true')
+
+      - script: ls -l
```
Comment (eman168) [inline azure-pipelines.yml]: is this required or just for debug?

Comment (kwan-nomensa) [reply azure-pipelines.yml]: Will take out, thx.

File: azure-pipelines.yml (hunk 1 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml:2]: Trigger manually when needed for now

Comment (timnaish-insolvency) [reply azure-pipelines.yml:2]: Then it's best to remove all triggers. `i.e. trigger: none`

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
+trigger:
+- release/*
+  
+pool:
+  vmImage: windows-latest
+
+parameters:
+- name: doPublish
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: This isn't really serving a useful purpose

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  default: false
+  type: boolean
+- name: toDeploy
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: This isn't required - you can simply deselect the deploy stage if you don't want to deploy

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+          arguments: '--configuration $(buildConfiguration) --collect "Code coverage"'
+
+      - task: DotNetCoreCLI@2
+        displayName: 'Publish Project'
+        inputs:
+          command: 'publish'
+          publishWebProjects: false
+          projects: '**/*.csproj'
+          arguments: '--configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)'
+        condition: eq('${{ parameters.doPublish}}', 'true')
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: No need to have a condition on this - we'd always want to make sure the publish step works, even if we don't save the resulting artefact.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+      - task: PublishPipelineArtifact@1
+        displayName: 'Publish Artifact'
+        inputs:
+          targetPath: '$(Build.ArtifactStagingDirectory)' 
+          artifactName: 'INSS.FIP'
+        condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+
+- stage: DeployDev
+  displayName: 'Deploy to Dev'
+  condition: and(and(succeeded(), ne(variables['Build.Reason'], 'PullRequest')), eq('${{ parameters.toDeploy}}', 'true'))
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: Don't need to use the toDeploy parameter

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+      parameters:
+        environment: 'Dev'
+        phaseBanner: $(PhaseBanner)
+        environmentBannerText: $(EnvironmentBannerText)
+        fipApiConnectorClientOptions__BaseAddress: $(FipApiConnectorClientOptions__BaseAddress)
+        iirwebdbContextConnectionString: $(IirwebdbContextConnectionString)
+        openApi__Version: $(OpenApi__Version)
+        openApi__DocVersion: $(OpenApi__DocVersion)
+        openApi__DocTitle: $(OpenApi__DocTitle)
+        appZipFile: 'INSS.FIP.zip'
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: This is always the same, so doesn't seem much point in supplying it every time

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        funcZipFile: 'INSS.FIP.FnApp.zip'
```
Comment (timnaish-insolvency) [inline azure-pipelines.yml]: Always the same

**Review by timnaish-insolvency — CHANGES_REQUESTED** [review]
A few optimisations suggested

Ken: implemented with last commit to branch

File: azure-pipelines.yml (hunk 3 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Removed in latest commit

File: azure-pipelines.yml (hunk 4 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Removed in latest commit

File: azure-pipelines.yml (hunk 5 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Removed in latest commit

File: azure-pipelines.yml (hunk 6 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Removed in latest commit

File: azure-pipelines.yml (hunk 7 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Removed in latest commit

File: azure-pipelines.yml (hunk 8 above)
Comment (kwan-nomensa) [reply azure-pipelines.yml]: Used default values in deploy template instead, no more pass-in parameters

**Review by timnaish-insolvency — CHANGES_REQUESTED** [review]
Currently the front end and back end are in two different app plans. This is not required - App Plans aren't a security construct, like a traditional server, but instead are resource allocations. Therefore there is no reason on such a small project to not put all the App Services/Functions in the same App Plan

**Review by murali-ins — APPROVED** [review]

**Review by timnaish-insolvency — APPROVED** [review]

**Pull request merged**